### DISCUSSION TITLE
Provide more info when migrations fail

### DISCUSSION THREAD

**Discussion by jwpjrdev** [diesel-rs/diesel#2907] [Ideas]
I just ran into an issue with a migration of mine that looks like this:
```sql
DROP EXTENSION pgcrypto
CREATE EXTENSION pgcrypto

CREATE TABLE users (
    id uuid DEFAULT gen_random_uuid (),
    username VARCHAR NOT NULL,
    password VARCHAR NOT NULL,services like Google
    email VARCHAR NOT NULL,
    email_verified BOOLEAN NOT NULL DEFAULT 'f',
    PRIMARY KEY (id)
)
```

Diesel returned this after I ran `diesel migration run`:

```
Running migration 2021-09-21-164249_create_users
Executing migration script /workspace/wishlist/server/migrations/2021-09-21-164249_create_users/up.sql
Failed with: syntax error at or near "CREATE"
```

In this situation, not very informative. Would be nice to see which line the issue was on or even where on the line it occurred.

**Comment by jwpjrdev** [comment]
Actually, it looks like an issue for this has already been created: https://github.com/diesel-rs/diesel/issues/2378

**Comment by jwpjrdev** [comment]
Aaaaand down the rabbit hole of Rust error handling trying

**Comment by jwpjrdev** [comment]
This is a lot more complex than I previously thought; to do this, it looks like you would have to have more information returned from `libpq` like where in the provided string the error occurred so it could be traced to a file in diesel.
I don't know C so if someone could take a look at this it would be greatly appreciated.

**Reply by weiznich** [reply]
https://github.com/diesel-rs/diesel/pull/2652 Implements already an interface to get more information about the exact location where an error occurred. So what's missing is some additional handling for the error [here](https://github.com/diesel-rs/diesel/blob/a51ebbd712afea1218b3ff417a24509ebab3674a/diesel_cli/src/main.rs#L205).

**Reply by jwpjrdev** [reply]
Ooh, that's pretty cool. I might be able to do that.

**Reply by weiznich** [reply]
If you need any other help there feel free to ask. We are generally happy to help people implementing features :wink:
